enterprise/frontend/src/metabase-enterprise/metabot/state/reducer.ts

```diff
@@ -75,6 +91,23 @@ export const metabot = createSlice({
     toolCallEnd: (state) => {
       state.activeToolCall = undefined;
     },
+    // NOTE: this reducer fn should be made smarter if/when we want to have
+    // metabot's `state` object be able to remove / forget values. currently
+    // we do not rewind the state to the point in time of the original prompt
+    // so if / when this becomes expected, we'll need to do some extra work here
+    // NOTE: this doesn't work in non-streaming contexts right now
```

Comment:
I tried to avoid the work of manually manipulate this history in the non-streamed UI. If we need more time to have confidence in streaming then happy to implement this if folks want.